.github/workflows/ci-app-tests.yml

```diff
@@ -38,14 +42,28 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
-      # TODO: use replace_oldest_ver() instead
+      - name: basic setup
+        run: pip install -q -r .actions/requirements.txt
+
       - name: Set min. dependencies
-        if: matrix.requires == 'oldest'
+        if: ${{ matrix.requires == 'oldest' }}
+        run: python .actions/assistant.py replace_oldest_ver
+      - run: python -c "print('TORCH_URL=https://download.pytorch.org/whl/' + str('test/cpu/torch_test.html' if '${{matrix.release}}' == 'pre' else 'cpu/torch_stable.html'))" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Install package
+        env:
+          PACKAGE_NAME: ${{ matrix.pkg-name }}
         run: |
-          for fpath in ('requirements/app/base.txt', 'requirements/app/test.txt'):
-              req = open(fpath).read().replace('>=', '==')
-              open(fpath, 'w').write(req)
-        shell: python
+          pip install -e . pytest --upgrade --find-links ${TORCH_URL}
```

Comment:
Why do we install pytest here? Shouldn't we do it through our requirements?

Reply:
I just want to install pyets so we can test the package itself - a situation you install the package without any extras...
Our test.txt already includes some other packages like cloud, pandas, etc.